### PR TITLE
Makefile: change make test to run just unit tests

### PR DESCRIPTION
**Pull request by jmrodri**
While working on the operator-sdk, I often type make test expecting to
run unit tests to do a quick verification. Sadly it would run ALL tests
causing a very long turn around time.

I updated the docs to let folks know to run make test for unit tests.
And removed the unit.sh script as it wasn't doing anything helpful.

### CODE REVIEW

**Comment by jmrodri** [comment]
```
[jesusr@speed3 operator-sdk{make-test}]$ make clean
[jesusr@speed3 operator-sdk{make-test}]$ make test
?       github.com/operator-framework/operator-sdk/commands/operator-sdk        [no test files]
ok      github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd    0.039s
?       github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/add        [no test files]
?       github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/completion [no test files]
?       github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/generate   [no test files]
?       github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/generate/internal  [no test files]
?       github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/olm-catalog        [no test files]
?       github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/run        [no test files]
?       github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/scorecard  [no test files]
?       github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/test       [no test files]
?       github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/up [no test files]
?       github.com/operator-framework/operator-sdk/pkg/ansible  [no test files]
ok      github.com/operator-framework/operator-sdk/pkg/ansible/controller       0.032s
ok      github.com/operator-framework/operator-sdk/pkg/ansible/controller/status        0.005s
?       github.com/operator-framework/operator-sdk/pkg/ansible/events   [no test files]
?       github.com/operator-framework/operator-sdk/pkg/ansible/flags    [no test files]
?       github.com/operator-framework/operator-sdk/pkg/ansible/operator [no test files]
?       github.com/operator-framework/operator-sdk/pkg/ansible/paramconv        [no test files]
ok      github.com/operator-framework/operator-sdk/pkg/ansible/proxy    0.033s
?       github.com/operator-framework/operator-sdk/pkg/ansible/proxy/controllermap      [no test files]
?       github.com/operator-framework/operator-sdk/pkg/ansible/proxy/kubeconfig [no test files]
?       github.com/operator-framework/operator-sdk/pkg/ansible/proxy/requestfactory     [no test files]
ok      github.com/operator-framework/operator-sdk/pkg/ansible/runner   0.009s
?       github.com/operator-framework/operator-sdk/pkg/ansible/runner/eventapi  [no test files]
?       github.com/operator-framework/operator-sdk/pkg/ansible/runner/fake      [no test files]
?       github.com/operator-framework/operator-sdk/pkg/ansible/runner/internal/inputdir [no test files]
?       github.com/operator-framework/operator-sdk/pkg/helm     [no test files]
?       github.com/operator-framework/operator-sdk/pkg/helm/client      [no test files]
?       github.com/operator-framework/operator-sdk/pkg/helm/controller  [no test files]
ok      github.com/operator-framework/operator-sdk/pkg/helm/engine      0.071s
?       github.com/operator-framework/operator-sdk/pkg/helm/flags       [no test files]
ok      github.com/operator-framework/operator-sdk/pkg/helm/internal/types      0.008s
?       github.com/operator-framework/operator-sdk/pkg/helm/release     [no test files]
?       github.com/operator-framework/operator-sdk/pkg/helm/watches     [no test files]
?       github.com/operator-framework/operator-sdk/pkg/internal/flags   [no test files]
ok      github.com/operator-framework/operator-sdk/pkg/k8sutil  0.030s
?       github.com/operator-framework/operator-sdk/pkg/leader   [no test files]
?       github.com/operator-framework/operator-sdk/pkg/log/zap  [no test files]
?       github.com/operator-framework/operator-sdk/pkg/metrics  [no test files]
?       github.com/operator-framework/operator-sdk/pkg/predicate        [no test files]
ok      github.com/operator-framework/operator-sdk/pkg/ready    0.004s
ok      github.com/operator-framework/operator-sdk/pkg/scaffold 1.333s
?       github.com/operator-framework/operator-sdk/pkg/scaffold/ansible [no test files]
?       github.com/operator-framework/operator-sdk/pkg/scaffold/helm    [no test files]
?       github.com/operator-framework/operator-sdk/pkg/scaffold/input   [no test files]
ok      github.com/operator-framework/operator-sdk/pkg/scaffold/olm-catalog     0.058s
?       github.com/operator-framework/operator-sdk/pkg/test     [no test files]
?       github.com/operator-framework/operator-sdk/pkg/test/e2eutil     [no test files]
?       github.com/operator-framework/operator-sdk/pkg/tls      [no test files]
```

**Comment by jmrodri** [comment]
this also makes `make all` more useful since it will run format, then test, then build the operator-sdk binary.

File: doc/dev/testing/travis-build.md [hunk 1]
```diff
@@ -112,7 +112,6 @@ The markdown test does not create a new cluster and runs in a barebones travis V
 [pr-builds]: https://travis-ci.org/operator-framework/operator-sdk/pull_requests
 [script]: ../../../hack/ci/setup-openshift.sh
 [sanity]: ../../../hack/tests/sanity-check.sh
-[unit]: ../../../hack/tests/unit.sh
```
Comment (LiliC) [inline doc/dev/testing/travis-build.md]: If you removed the link to the file, you also need to remove the reference to this link. Change the readme https://github.com/operator-framework/operator-sdk/blob/b23e3e145904bdef0124830d8372d5d53a59b771/doc/dev/testing/travis-build.md#go-tests

**Comment by shawn-hurley** [comment]
+100 from me

File: doc/dev/testing/travis-build.md (hunk 1 above)
Comment (jmrodri) [reply doc/dev/testing/travis-build.md]: @LiliC updated the doc, thanks for catching that.

File: doc/dev/testing/travis-build.md [hunk 2]
```diff
@@ -32,7 +32,8 @@ The Go, Ansible, and Helm tests then differ in what tests they run.
     2. Check that all source files have a license.
     3. Check that all error messages start with a lower case alphabetical character and do not end with punctuation, and log messages start with an upper case alphabetical character.
     4. Make sure the repo is in a clean state (this is particularly useful for making sure the `Gopkg.lock` file up to date after `dep ensure`).
-2. Run [unit tests][unit].
+2. Run unit tests.
+    1. Run `make test` or `go test -count=1 -short ./commands/... ./pkg/...`.
```
Comment (LiliC) [inline doc/dev/testing/travis-build.md]: Nit: Think the go test command is not needed, as user can see that in the Makefile target.

Comment (jmrodri) [reply doc/dev/testing/travis-build.md]: Removed the go command.

**Pull request merged**